### PR TITLE
CVSL-1707 add client calls and variables to entities

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceRepository.kt [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 import org.springframework.data.jpa.repository.JpaRepository
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor
 import org.springframework.stereotype.Repository
+import java.time.LocalDateTime
 
 @Repository
 interface LicenceRepository : JpaRepository<Licence, Long>, JpaSpecificationExecutor<Licence> {
   fun findByPrisonNumber(prisonNumber: String, pageable: Pageable): Page<Licence>
   fun findAllByPrisonNumber(prisonNumber: String): List<Licence>
+  fun findAllByDeletedAtAndIdGreaterThanLastProcessedAndOrderByIdAsc(deletedAt: LocalDateTime?, lastProcessed: Long, limit: Limit): Page<Licence>
```
Comment (louisenorris) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceRepository.kt:15]: Not 100% sure if this is the correct way to do this, following the use of [keywords supported for JPA](https://docs.spring.io/spring-data/jpa/reference/jpa/query-methods.html) 👀

Comment (louisenorris) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceRepository.kt:15]: Hoping it does all the following: `select * from licence where id > ${lastIdProcessed} and deleted is null order by id asc limit ${batchSize}` 😅

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceRepository.kt:15]: when you write the integration test you should be able to see the generated sql in the logs of the test run

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceRepository.kt:15]: (if it doesn't you could write a custom query)

**Review by andrewrlee — COMMENTED** [review]
LGTM

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**